eventio/iact/objects.py

```diff
@@ -145,12 +145,20 @@ def parse(self):
         n_arrays = read_int(self)
         time_offset = read_float(self)
 
-        return time_offset, read_array(
+        n_par = len(self.dtypes[self.header.version])
```

Comment:
Can you replace `dytpes` above with just `columns = {0: ['x', 'y'], 1: ['x', 'y', 'weight']}` then use:

```
        columns = self.columns[self.header.version]
        n_columns = len(columns)

        # each of the columns are stored after another
        offsets = read_array(self, count=n_arrays * n_columns, dtype=np.float32)
        offsets = offsets.reshape((n_columns, n_arrays))
        offsets = np.core.records.fromarrays(offsets, names=columns)

        return time_offset, offsets
```

I think that makes it clearer, 
```

Reply:
Sure, I just changed it